include/deal.II/grid/grid_tools.h

```diff
@@ -748,6 +748,28 @@ namespace GridTools
                                  const std::vector<bool>      &marked_vertices = std::vector<bool>());
 
   /**
+   * A version of the previous function that exploits an already constructed
+   * map between vertices and cells, constructed using the function
+   * GridTools::vertex_to_cell_map, a map of vertex_to_cell_centers, obtained
+   * through GridTools::vertex_to_cells_center_directions, and a guess `current_cell`.
+   *
+   * @author Luca Heltai, Rene Gassmoeller, 2017
+   */
+  template <int dim, template <int, int> class MeshType, int spacedim>
+#ifndef _MSC_VER
+  std::pair<typename MeshType<dim, spacedim>::active_cell_iterator, Point<dim> >
+#else
+  std::pair<typename dealii::internal::ActiveCellIterator<dim, spacedim, MeshType<dim, spacedim> >::type, Point<dim> >
+#endif
+  find_active_cell_around_point (const Mapping<dim,spacedim>                                                          &mapping,
+                                 const MeshType<dim,spacedim>                                                         &mesh,
+                                 const Point<spacedim>                                                                &p,
+                                 const typename MeshType<dim, spacedim>::active_cell_iterator                         &current_cell,
```

Comment:
I think we should rename this to `cell_hint` or something similar. `current_cell` seems slightly confusing.